src/engine/renderer/tr_shader.cpp

```diff
@@ -2895,26 +2929,79 @@ static bool ParseStage( shaderStage_t *stage, const char **text )
 		{
 			ParseExpression( text, &stage->fresnelBiasExp );
 		}
-		// normalScale <float> <float> <float>
+		// normalFormat <+/-> <+/-> <+/->
 		//
-		// for compatibility purpose, ioq3gl2 syntax is supported:
-		// normalScale <float> <float> 
+		// Describes the normal map format for the the given normal map file.
 		//
-		// scale normalMap channel, can flip channels with negative values
-		//
-		// expects OpenGL scale as default: 1 1 1
+		// Since this parser assumes DirecX format to keep
+		// compatibility with materials created for ioquake3
+		// normalFormat must be set to: + + +
+		// to load normal maps in OpenGL format.
 		//
 		// OpenGL format is described there:
 		//     https://github.com/KhronosGroup/glTF/tree/2.0/specification/2.0#materialnormaltexture
 		//
 		// > The normal vector uses OpenGL conventions where +X is right, +Y is up, and +Z points toward the viewer.
 		//
-		// examples of scales:
+		// examples of formats:
+		//
+		// + + + OpenGL format
+		// + - + DirectX format with reverted green channel (reverted Y component)
+		// - + + weird other format with reverted red channel (reverted X component)
+		//
+		// Unlike normalScale that can be used to revert channels by setting them
+		// negative, the normalFormat keyword is meant to be engine agnostic:
+		// engines implementing either OpenGL or DirectX format internally
+		// are expected to do the required normal channel flip themselves.
+		else if ( !Q_stricmp( token, "normalFormat" ) )
+		{
+			// replace existing normal format in all case
+			stage->hasNormalFormat = true;
+
+			for ( int i = 0; i < 3; i++ )
+			{
+				token = COM_ParseExt2( text, false );
+
+				if ( token[ 0 ] == '+' )
+				{
+					Log::Warn("Found + normalFormat parm in shader '%s'", shader.name );
```

Comment:
Why is this a warning? Isn't that the expected input?

Reply:
Ah, sorry, that was debug code to follow what I was doing, will remove it. :-)